### PR TITLE
The message type + codes for the audit stream

### PR DESCRIPTION
**Pull request by alicja-gruzdz**
The audit stream carries information on all operations performed in the platform.

The message codes included are for the basic operations; 
- operations on Instance,
- operations on Sequence,
- operations on Topics,
- internal system operations.

Suggestions on the improvements of the message code created are very welcome. Perhaps for the codes we should use string and create a hash from the request method name (can be digits) + the part of URL (after removing the base and the IDs from the URL). 
The current codes are based on @patuwwy suggestion to have the first digit indicating a category of operations (Instance, Sequence, etc.) and the second digit for the method. However, the last digits would still need to be known. 

There is only one message type proposed with all required attributes listed without nesting. It's not the neatest modelling, but otherwise we would need to maintain a long list of various combinations of message codes and attributes.

[EDIT]: After debugging the Auditor service with @patuwwy I adopted his status codes convention which can be found in https://github.com/scramjetorg/transform-hub/pull/461/files#diff-10ce2297b522d8ec475f58fc3b0c9022d899a48ff62cb9430c28f503ccf36389

### CODE REVIEW

**Comment by alicja-gruzdz** [comment]
After the conversation with @patuwwy we decided to focus only on the API operation registration for now. When all is working we will be extended the message in attributes and codes as they are required.

File: packages/symbols/src/op-record-code.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    GET_INSTANCE_EVENT_ONCE = 8010,
+    POST_INSTANCE_INPUT = 8100,
+    POST_INSTANCE_MONITORING_RATE = 8101,
+    POST_INSTANCE_EVENT = 8102,
+    POST_STOP_INSTANCE = 8103,
+    POST_KILL_INSTANCE = 8104,
+
+    /**
+     * The message codes related to the operation on a Sequence.
+     * They are trigger in response for the user operations performed by the API.
```
Comment (a-tylenda) [inline packages/symbols/src/op-record-code.ts:27]:
```suggestion
     * They are triggered in response to the user operations performed by the API.
```

File: packages/symbols/src/op-record-code.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
+export enum OpRecordCode {
+
+    /**
+     * The message codes related to the operation on an Instance.
+     * They are trigger in response for the user operations performed by the API.
```
Comment (a-tylenda) [inline packages/symbols/src/op-record-code.ts:5]:
```suggestion
     * They are triggered in response to the user operations performed by the API.
```

File: packages/symbols/src/op-record-code.ts (hunk 1 above)
Comment (a-tylenda) [inline packages/symbols/src/op-record-code.ts:27]:
```suggestion
     * They are triggered in response to the user operations performed by the API.
```

File: packages/symbols/src/op-record-code.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+     */
+
+    GET_SEQUENCE = 9000,
+    DELETE_SEQUENCE = 9200,
+    POST_SEQUENCE = 9100,
+    POST_START_INSTANCE = 9101,
+
+    /**
+     * The message codes related to the operation on Topics.
+     * They are trigger in response for the user operations performed by the API.
```
Comment (a-tylenda) [inline packages/symbols/src/op-record-code.ts:37]:
```suggestion
     * They are triggered in response to the user operations performed by the API.
```

**Review by a-tylenda — APPROVED** [review]

File: packages/symbols/src/op-state.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+export enum OpState {
+
```
Comment (patuwwy) [inline packages/symbols/src/op-state.ts:2]:
```suggestion
```

File: packages/types/src/messages/op-record.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
+import { OpRecordCode } from "@scramjet/symbols";
+
+export type OpRecord = {
+
```
Comment (patuwwy) [inline packages/types/src/messages/op-record.ts:4]:
```suggestion
```

File: packages/types/src/messages/op-record.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    /**
+    * The operation state from the OpState enumeration.
+    */
+    opState: string;
+
+    /**
+    * The generated unique request ID for operations coming from the API.
+    */
+
```
Comment (patuwwy) [inline packages/types/src/messages/op-record.ts:18]:
```suggestion
```

File: packages/types/src/messages/op-record.ts [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    /**
+    * The data received in the request stream
+    */
+    rx?: number;
+
+    /**
+    * An instance of the object which the operation concerns, e.g. Host.
+    */
+    objectId: string;
+
```
Comment (patuwwy) [inline packages/types/src/messages/op-record.ts:45]:
```suggestion
```

Comment (alicja-gruzdz) [reply packages/types/src/messages/op-record.ts:45]: thanks :)

File: packages/symbols/src/op-record-code.ts (hunk 1 above)
Comment (alicja-gruzdz) [reply packages/symbols/src/op-record-code.ts:27]: thanks Aga!

**Review by MichalCz — APPROVED** [review]

**Pull request merged**